Property 'toResult' is missing in type 'IMaybe<T>' but required in type 'Maybe<T>'

When I use `none` function, it return a "IMaybe" interface instead of "Maybe", but there is no "toResult" in "IMaybe", so, it is not a "Maybe".

If I set 'strict' in  tsconfig.json to 'false', errors gone. 

So is it a bug?

```
src/domain/tree.ts:42:10 - error TS2741: Property 'toResult' is missing in type 'IMaybe<T>' but required in type 'Maybe<T>'.

42     else return none()
            ~~~~~~~~~~~~~

  node_modules/typescript-monads/maybe/maybe.d.ts:27:5
    27     toResult<E>(error: E): IResult<T, E>;
           ~~~~~~~~
    'toResult' is declared here.
```

--- a/src/maybe/maybe.interface.ts
+++ b/src/maybe/maybe.interface.ts
@@ -1,4 +1,5 @@
 import { IMonad } from '../monad/monad.interface'
+import { IResult } from '../result/result.interface'
 
 /**
  * Define a contract to unwrap Maybe object
@@ -118,4 +119,6 @@ export interface IMaybe<T> extends IMonad<T> {
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   apply(maybe: IMaybe<ReturnType<T extends (...args: any) => any ? T : any>>): IMaybe<T>
+
+  toResult<E>(error: E): IResult<T, E>
 }

